client/trino-client/src/main/java/io/trino/client/QueryData.java

```diff
@@ -17,8 +17,17 @@
 
 import java.util.List;
 
+/**
+ * Used for representing both raw JSON values and spooled metadata.
+ */
 public interface QueryData
 {
+    @Deprecated
     @Nullable
-    Iterable<List<Object>> getData();
+    default Iterable<List<Object>> getData()
+    {
+        throw new UnsupportedOperationException("getData() is deprecated for removal, use StatementClient.currentRows() instead");
```

Comment:
Default implementation which throws makes no sense.
* If method is used then it means you need specific implementation of this method in implementations of `QueryData`.
* If it is no longer used just remove it from interface and implementatins.

Reply:
I want this to be backward compatible for some time hence the default implementation